src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -5186,149 +5180,150 @@ def remove_nic_ip_config_inbound_nat_rule(
+        _handle_plural_or_singular(args, "source_address_prefixes", "source_address_prefix")
+        _handle_plural_or_singular(args, "source_port_ranges", "source_port_range")
+        # handle application security groups
+        if has_value(args.destination_asgs):
+            args.destination_application_security_groups = [{"id": asg_id} for asg_id in args.destination_asgs]
+            if has_value(args.destination_address_prefix):
+                args.destination_address_prefix = None
+        if has_value(args.source_asgs):
+            args.source_application_security_groups = [{"id": asg_id} for asg_id in args.source_asgs]
+            if has_value(args.source_address_prefix):
+                args.source_address_prefix = None
 
-class NsgRuleUpdate(_NsgRuleUpdate):
 
+class NSGRuleUpdate(_NSGRuleUpdate):
     @classmethod
     def _build_arguments_schema(cls, *args, **kwargs):
         from azure.cli.core.aaz import AAZListArg, AAZResourceIdArg, AAZListArgFormat, AAZResourceIdArgFormat
 
         class EmptyListArgFormat(AAZListArgFormat):
             def __call__(self, ctx, value):
-                data = value._data
-                if has_value(data) and len(data) == 1 and data[0] == "":
-                    logger.warning("It's recommended to detach it by null, empty string (\"\") will be deprecated.")
-                    value._data = None
-                return super().__call__(ctx, value)
-
-        class EmptyResourceIdArgFormat(AAZResourceIdArgFormat):
-            def __call__(self, ctx, value):
-                if value._data == "":
+                if value.to_serialized_data() == [""]:
                     logger.warning("It's recommended to detach it by null, empty string (\"\") will be deprecated.")
                     value._data = None
                 return super().__call__(ctx, value)
 
         args_schema = super()._build_arguments_schema(*args, **kwargs)
         args_schema.destination_asgs = AAZListArg(
-            options=['--destination-asgs'],
+            options=["--destination-asgs"],
             arg_group="Destination",
+            help="Space-separated list of application security group names or IDs. Limited by backend server, "
+                 "temporarily this argument only supports one application security group name or ID.",
             nullable=True,
-            help="Space-separated list of application security group names or supports one application security group name or ID. Use null to detach it.",
-            fmt=EmptyListArgFormat()
+            fmt=EmptyListArgFormat(),
         )
         args_schema.destination_asgs.Element = AAZResourceIdArg(
             nullable=True,
-            fmt=EmptyResourceIdArgFormat(
-                template="/subscriptions/{subscription}/resourceGroups/{resource_group}/providers/Microsoft.Network/applicationSecurityGroups/{}"
-            )
+            fmt=AAZResourceIdArgFormat(
```

Comment:
Why it's changed from `EmptyResourceIdArgFormat` to `AAZResourceIdArgFormat`. It's a breaking change.